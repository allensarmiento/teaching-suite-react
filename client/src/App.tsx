import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { Classroom } from './components/classroom/classroom.component';
import { HomePage } from './components/home-page/home-page.component';
import {
  TopNavigation
} from './components/top-navigation/top-navigation.component';
import './App.css';

function App() {
  return (
    <div className="App">
      <TopNavigation title="Teaching Suite" username="Test" />
      <BrowserRouter>
        <Switch>
          <Route path="/classroom">
            <Classroom />
          </Route>
          <Route path="/">
            <HomePage />
          </Route>
        </Switch>
      </BrowserRouter>
    </div>
  );
}

export default App;
