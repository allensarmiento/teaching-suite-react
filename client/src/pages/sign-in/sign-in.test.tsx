import { mount, shallow } from 'enzyme';
import SignIn from './sign-in.component';
import { CurrentUser } from '../../App';
import BlueGreyButton from '../../core/blue-grey-button/blue-grey-button.component';

const signinFn = jest.fn();
const email = 'test@test.com';
const password = '12345';

it('renders sign in form component if user not signed in', () => {
  expect(shallow(<SignIn currentUser={null} signin={signinFn} />))
    .toMatchSnapshot();
});

it('renders submit button', () => {
  const wrapper = shallow(<SignIn currentUser={null} signin={signinFn} />);

  const button = wrapper.find(BlueGreyButton);
  expect(button.prop('type')).toEqual('submit');
});

it('redirects to home page if user is signed in', () => {
  const mockCurrentUser: CurrentUser = {
    id: 1,
    email: 'test@test.com',
    role: 'student',
  };

  expect(shallow(<SignIn currentUser={mockCurrentUser} signin={signinFn} />))
    .toMatchSnapshot();
});

it('updates state on input change', () => {
  const wrapper = shallow(<SignIn currentUser={null} signin={signinFn} />);

  const emailInput = wrapper.find('[name="email"]');
  emailInput.simulate('change', { target: {
    value: email,
    name: 'email',
  }});

  const passwordInput = wrapper.find('[name="password"]');
  passwordInput.simulate('change', { target: {
    value: password,
    name: 'password',
  }});
  
  expect(wrapper.state()).toEqual({ email, password });
});

it('calls sign in function on form submit', () => {
  const wrapper = mount(<SignIn currentUser={null} signin={signinFn} />);

  const emailInput = wrapper.find('[name="email"]');
  emailInput.simulate('change', { target: {
    value: email,
    name: 'email',
  }});

  const passwordInput = wrapper.find('[name="password"]');
  passwordInput.simulate('change', { target: {
    value: password,
    name: 'password',
  }});

  wrapper.update();

  const form = wrapper.find('form');
  form.simulate('submit');
  expect(signinFn).toHaveBeenCalledTimes(1);
  expect(signinFn).toHaveBeenCalledWith(email, password);
});
