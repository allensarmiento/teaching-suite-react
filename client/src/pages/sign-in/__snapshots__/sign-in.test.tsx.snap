// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redirects to home page if user is signed in 1`] = `
<Redirect
  to="/"
/>
`;

exports[`renders sign in form component if user not signed in 1`] = `
<div
  className="signin"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <h2
      className="title"
    >
      Sign In
    </h2>
    <input
      className="input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
      value=""
    />
    <input
      className="input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <BlueGreyButton
      className="button"
      type="submit"
    >
      Sign In
    </BlueGreyButton>
  </form>
</div>
`;
